### ISSUE TITLE
Cannot set "icon" on "[object Object]". "[object Object]" is read-only.

### ISSUE THREAD

**Issue by jduelfer** [pozil/sfdc-ui-lookup-lwc#11]
When not specifying an icon, an error is throw due to the Locker Service (or however LWC doesn't let us mutate data that comes from Apex).

Stacktrack:
```
Error: Invalid mutation: Cannot set "icon" on "[object Object]". "[object Object]" is read-only.
    at ReadOnlyHandler.set (aura_proddebug.js:4914)
    at UnfilteringProxyHandlerForObject.set (aura_proddebug.js:12356)
    at eval (lookup.js:326)
    at Proxy.map (<anonymous>)
    at Lookup.setSearchResults (lookup.js:324)
    at callHook (aura_proddebug.js:12855)
    at HTMLElement.<anonymous> (aura_proddebug.js:5326)
    at Proxy.value (aura_proddebug.js:17845)
    at eval (pdwFilter.js:54)
```

It happens in this function:
```javascript
@api
setSearchResults(results) {
    this.searchResults = results.map(result => {
        if (typeof result.icon === 'undefined') {
            result.icon = 'standard:default';
        }
        return result;
    });
}
```

Instead of defaulting this in the front-end, it _could_ be done in the `LookupSearchResult.cls`. Since that is a sort of bean, that might not be the cleanest solution, but it would be _a_ fix.

So in my org, I temporarily have
```java
@AuraEnabled
public String getIcon() {
    return icon != null ? icon : 'standard:default';
}
```

**Comment by pozil** [comment]
Hi @jduelfer, thanks for reporting this issue.

I fixed it in release 1.0.4 on [the component's JS](https://github.com/pozil/sfdc-ui-lookup-lwc/blob/a83c371c866311f223f446ab81a9a7ecebfc0361/src/main/default/lwc/lookup/lookup.js#L29) by cloning the search result when there is no icon.

**Comment by jduelfer** [comment]
That's a much cleaner solution, so I'll pull those changes. Thanks for the fix.
